File: gems/decomplex/examples/zig/implicit-control-flow.zig
const FlowExample = struct {
    status: i32,
    valid: bool,
    done: bool,

    pub fn prepare(self: *FlowExample) void { self.status = 1; }
    pub fn validate(self: *FlowExample) void { self.valid = self.status == 1; }
    pub fn commit(self: *FlowExample) void { self.done = self.valid; }

    pub fn ok1(self: *FlowExample) void { self.prepare(); self.validate(); self.commit(); }
    pub fn ok2(self: *FlowExample) void { self.prepare(); self.validate(); self.commit(); }
    pub fn ok3(self: *FlowExample) void { self.prepare(); self.validate(); self.commit(); }
    pub fn ok4(self: *FlowExample) void { self.prepare(); self.validate(); self.commit(); }
    pub fn drift(self: *FlowExample) void { self.validate(); self.prepare(); self.commit(); }
};

File: gems/decomplex/examples/rust/implicit-control-flow.rs
struct FlowExample {
    status: i32,
    valid: bool,
    done: bool,
}

impl FlowExample {
    fn prepare(&mut self) { self.status = 1; }
    fn validate(&mut self) { self.valid = self.status == 1; }
    fn commit(&mut self) { self.done = self.valid; }

    fn ok1(&mut self) { self.prepare(); self.validate(); self.commit(); }
    fn ok2(&mut self) { self.prepare(); self.validate(); self.commit(); }
    fn ok3(&mut self) { self.prepare(); self.validate(); self.commit(); }
    fn ok4(&mut self) { self.prepare(); self.validate(); self.commit(); }
    fn drift(&mut self) { self.validate(); self.prepare(); self.commit(); }
}
